### DISCUSSION TITLE
Language show as 99% Javascript although there are Typescript Files

### DISCUSSION THREAD

**Discussion by Lucaslah** [github-linguist/linguist#5348] [Classification] [answered]
## Problem description
Language showing as 99% javascript even though there are lots of Typescript Files, but when I go to see what files are what language it shows lots of the files are Typescript. It should at least show some Typescript in the repo.

### URL of the affected repository:
https://github.com/CSSUDII/public

### Last modified on:
2021-05-1

### Expected language:
Mostly Typescript with some Javascript

### Detected language:
99.4% Javascript and 0.6% Other

### Screenshots
![268D71AF-D1BA-471B-A625-D1166297D8C7](https://user-images.githubusercontent.com/47047328/116781251-de580b00-aad5-11eb-836f-caa02e035d82.jpeg)
![5498375C-F12E-4E95-B578-E4CF374A1E5D](https://user-images.githubusercontent.com/47047328/116781253-e0ba6500-aad5-11eb-9a6e-2fe3d5a92a1e.jpeg)

**Comment by lildude** [comment, marked as answer]
From the [How Linguist Works](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md) doc:

> The percentages are calculated based on the bytes of code for each language as reported by the List Languages API.

Looking at your repo, we can see your language breakdown by bytes of code is:

    JavaScript 2.84 MB
    TypeScript 13.4 KB
    HTML 1.68 KB
    CSS 531 Bytes
    Shell 35 Bytes

You can see your repo has significantly more JavaScript than anything else and it's all pretty much down to one massive file: https://github.com/CSSUDII/public/blob/main/bundle.js . This file alone is 2.82MB and to blame for the high proportion of JavaScript.

You'll need to use an override to hide this file from the stats if you don't want it to count.

**Reply by Lucaslah** [reply]
Ah, Okay, Thanks for that.
So I'll just add an override for that file.
